### ISSUE TITLE
Bug: missing property `fatalErrorCount` in ignored file results

### ISSUE THREAD

**Issue by fasttime** [eslint/eslint#15510]
### Environment

Node version: 17.3.1
npm version: 8.3.0
Local ESLint version: 8.6.0

### What parser are you using?

Default (Espree)

### What did you do?

**Code**

```js
// test.mjs
import assert from 'assert/strict';
import { ESLint } from 'eslint';

const eslint = new ESLint({ useEslintrc: false });
const results = await eslint.lintText('', { filePath: '/a/b/c/.ignored.js', warnIgnored: true });
assert.equal(results[0].fatalErrorCount, 0); // fails
```

**Output**

```
AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:

undefined !== 0
```

### What did you expect to happen?

Results produced by ignored files with the `warnIgnored` option should have a `fatalErrorCount` property like other results.
The assertion in the repro code should pass.

### What actually happened?

Results produced by ignored files have no `fatalErrorCount` property.

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by nzakas** [comment]
On my phone so unable to reproduce at the moment, but interestingly it does look like we aren't checking for `fatalErrorCount` in this test:

https://github.com/eslint/eslint/blob/808ad35f204c838cd5eb8d766807dc43692f42f9/tests/lib/eslint/eslint.js#L312

I can't see a code path that would add/remove `fatalErrorCount` anywhere, but I'll need to take a closer look when I'm back at the computer.

**Comment by aladdin-add** [comment]
Yes, I was able to repro. looks like a bug to me.

**Comment by nzakas** [comment]
Thanks @aladdin-add! @fasttime if you’d like to submit a pull request, we’d appreciate it.

**Comment by fasttime** [comment]
@nzakas Sure! Thanks for confirming this.
